### PR TITLE
searchBar-implementation

### PR DESCRIPTION
**Pull request by JaqKent**
funcionality of the search bar...part 1

### CODE REVIEW

File: src/components/searchBar/index.tsx [hunk 1]
```diff
@@ -1,14 +1,54 @@
+import { ProductsContext } from "context/Product";
+import { useContext, useState } from "react";
```
Comment (nathan-hyan) [inline src/components/searchBar/index.tsx]: React goes first always

File: src/components/searchBar/index.tsx [hunk 2]
```diff
@@ -1,14 +1,54 @@
...
 import styles from "./styles.module.scss";
 
 function SearchBar() {
+  const { products } = useContext(ProductsContext);
+  const [value, setValue] = useState("");
+
+  const onChange = (event: any) => {
```
Comment (nathan-hyan) [inline src/components/searchBar/index.tsx]: Don't any

**Review by nathan-hyan — CHANGES_REQUESTED** [review]
Great job, some comments

File: src/components/searchBar/index.tsx (hunk 1 above)
Comment (JaqKent) [reply src/components/searchBar/index.tsx]: done

File: src/components/searchBar/index.tsx (hunk 2 above)
Comment (JaqKent) [reply src/components/searchBar/index.tsx]: done

File: src/components/searchBar/index.tsx [hunk 3]
```diff
@@ -1,14 +1,56 @@
...
+  const { products } = useContext(ProductsContext);
+  const [value, setValue] = useState("");
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  const onSearch = (searchItem: string) => {
+    setValue(searchItem);
+    console.log("search", searchItem);
```
Comment (nathan-hyan) [inline src/components/searchBar/index.tsx]: Console log

File: src/components/searchBar/index.tsx [hunk 4]
```diff
@@ -1,14 +1,56 @@
...
+        </button>
+      </div>
+      <div className={styles.dropdown}>
+        {filteredProducts.map((item) => (
+          <div
+            onClick={() => onSearch(item.name)}
+            className={styles.dropdownRow}
+            key={item._id}
+          >
+            {" "}
```
Comment (nathan-hyan) [inline src/components/searchBar/index.tsx]: Remove this space if it isnt necessary

**Review by nathan-hyan — CHANGES_REQUESTED** [review]
Two comments left

**Pull request merged**